pkg/controller/opentelemetrycollector/reconcile_service.go

```diff
@@ -209,3 +228,37 @@ func serviceLabels(ctx context.Context, name string) map[string]string {
 	labels["app.kubernetes.io/name"] = name
 	return labels
 }
+
+func filterPort(candidate corev1.ServicePort, portNumbers map[int32]bool, portNames map[string]bool) *corev1.ServicePort {
+	if portNumbers[candidate.Port] {
+		return nil
+	}
+
+	// do we have the port name there already?
+	if portNames[candidate.Name] {
+		// there's already a port with the same name! do we have a 'port-%d' already?
+		fallbackName := fmt.Sprintf("port-%d", candidate.Port)
+		if portNames[fallbackName] {
+			// that wasn't expected, better skip this port
```

Comment:
We should log this. Should the last resort be a random name?

Reply:
Added a log statement for this, but this is such an edge case that I wouldn't generate a random name for it.